keel-ec2/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/AmazonSecurityGroupHandler.kt

```diff
@@ -124,18 +130,67 @@ class AmazonSecurityGroupHandler(
       .apply {
         typeMetadata = request.asset.typeMetadata
         spec = SecurityGroupProto.newBuilder()
-          .also {
-            it.name = name
-            it.accountName = accountName
-            it.region = region
-            it.vpcName = vpcId?.let { cloudDriverCache.networkBy(it).name }
-            it.description = description
+          .also { builder ->
+            builder.name = name
+            builder.accountName = accountName
+            builder.region = region
+            builder.vpcName = vpcId?.let { cloudDriverCache.networkBy(it).name }
+            builder.description = description
+            inboundRules.forEach { rule ->
```

Comment:
This whole bit with the inbound rules was missing so the current and desired states never match.

Reply:
🤦‍♂️ oops